File: Shitty FOrtune.py
def check(s):
    for i in range(0,len(s)):
        if(s[i]!=s[len(s)-i-1]):
            return False
    return True
t=int(input())
while(t>0):
    s=input().split(" ")
    k=int(k)
    if(len(s)%k==0):
        kk=len(s)//k
        flag=1
        for i in range(0,len(s)-kk,kk):
            if(check(s[i:i+kk])):
                     flag+=1

        if(flag==k):
            print("YES")
        else:
            print("NO")
    else:
        print("NO") 
    t-=1

File: Shitty Fortune.cpp
#include<bits/stdc++.h>
using namespace std;
int check(string s);
int main()
{
	long long int t;
	cin>>t;
	while(t--)
	{
		long long int k;
		string s;
		cin>>s>>k;
		if(s.length()%k==0)
		{
			long long int kk=s.length()/k;
			long long flag=0;
			for(long long int i=0;i<s.length();i+=kk)
			{
				if(check(s.substr(i,kk)))
				{
					flag+=1;
				}
			}
			if(flag==k)
			{
				cout<<"YES"<<endl;
				
			}
			else
			{
				cout<<"NO"<<endl;
			}
		}
		else
		{
			cout<<"NO"<<endl;
		}
	}
}
int check(string s)
{
	for(long long int i=0;i<s.length();i++)
	{
		if(s[i]!=s[s.length()-i-1])
		{
			return 0;
		}
	}
	return 1;
}
